plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1007,6 +1008,18 @@ public void testNullSensitiveTopNPushdown()
         }
     }
 
+    @Test
+    public void testPredicateModuloPushdown()
+    {
+        if (!hasBehavior(SUPPORTS_PREDICATE_MODULO_PUSHDOWN)) {
```

Comment:
Due to high number for connector behaviors, it's imperative that these declarations are self-tested. I.e. both "supports" and "does not support" should be verified. Otherwise, connector implementors will not know that they need to declare that give behavior is supported and it will not be tested.

Concrete example: PostgreSQL supports modulo pushdown, and automation should remind us to declare that in the PostgreSQL's connector test.

Reply:
added 
```
   if (!hasBehavior(SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN)) {
            assertThat(query("SELECT shippriority FROM orders WHERE shippriority % 4 = 0")).isNotFullyPushedDown(FilterNode.class);
            return;
        }
```
But didn't get fully what should be the node, I took for now FilterNode,
but can you please clarify is it correct way

Reply:
The easiest way is to change to `isFullyPushedDown`, see what the error shows the plan looks like and add the node (or sub-tree) just above TableScan as argument to `isNotFullyPushedDown`.

e.g. if connector doesn't support LIMIT pushdown but supports aggregation pushdown then a query like `SELECT regionkey, sum(nationkey) FROM (SELECT * FROM nation WHERE regionkey < 2 LIMIT 11) GROUP BY regionkey` will have plan like:

`... LimitNode ... -> TableScanNode`

That can be represented as `isNotFullyPushedDown(LimitNode.class)` or to be more precise `isNotFullyPushedDown(node(LimitNode.class, anyTree(node(TableScanNode.class))))` (i.e. a LimitNode followed by 1 or more nodes followed by a TableScanNode at the end).

Reply:
I made this test and see that last node is ScanFilter
So looks like isNotFullyPushedDown(FilterNode.class); is correct

Reply:
Yes, FilterNode is correct here.

ScanFilter = FilterNode + TableScanNode
ScanFilterProject = ProjectNode + FilterNode + TableScanNode.